src/main/java/com/speaker/service/MessageServiceImpl.java

```diff
@@ -22,22 +25,31 @@ public class MessageServiceImpl implements MessageService {
     private final MessageRepository messageRepository;
     private final AccountRepository accountRepository;
     private final MessageConvertor messageConvertor;
+    private final FieldValidators<MessageDTO> messageDTOValidators;
 
     @Override
     public List<Message> getMessagesByAccountId(int accountId) {
         return messageRepository.findAllByAccountId(accountId);
     }
 
-    @Override
-    public Response addMessage(List<MessageDTO> messageDTOs) {
+        @Override
+    public List<ValidatorError> addMessage(List<MessageDTO> messageDTOs) {
+        List<ValidatorError> validatorErrors = messageDTOs.stream()
+                .map(messageDTOValidators::validate)
+                .flatMap(Collection::stream)
+                .collect(Collectors.toList());
+        if (validatorErrors.size() > 0) {
+            return validatorErrors;
+        }
         List<Message> messages = messageDTOs.stream()
                 .map(this::convertToMessage)
                 .filter(Objects::nonNull)
                 .collect(Collectors.toList());
         if (messageDTOs.size() == messages.size()) {
-            return messageRepository.createMessages(messages) ? Response.TRUE : Response.FALSE;
+           messageRepository.createMessages(messages);
+           return List.of(new ValidatorError("messages  created", ""));
```

Comment:
`new ValidatorError("messages was created", "messageDTOs")`

Reply:
Comments accepted, corrections made. The message is extended